cat epilepsy_cohort.hg19_multianno.vcf |perl -lane 'if(/Func.refGene=exonic/){print $_}if(/Func.refGene=ncRNA_exonic/){print $_}if(/Func.refGene=splicing/){print $_}' |perl -lane 'if(/ExonicFunc.refGene=frameshift/){print $_}if(/ExonicFunc.refGene=nonframeshift/){print $_}if(/ExonicFunc.refGene=start/){print $_}if(/ExonicFunc.refGene=stop/){print $_}if(/ExonicFunc.refGene=nonsynonymous/){print $_}elsif(/Func.refGene=splicing/){print $_}' |sed 's/ /_/g' > GATK_Exon_filter.vcf
awk -f /public/group_share_data/fan_lab/suke/Epilepsy/VQSR/annovar/split_new.awk GATK_Exon_filter.vcf >GATK_Exon_filter_split.vcf     ## split_new.awk is provided as a separate file
cat epilepsy_cohort.vcf |awk '{if($0 ~ /^#CH/){print}}' >epilepsy_cohort_head.txt
cat epilepsy_cohort_head.txt GATK_Exon_filter_split.vcf >GATK_Exon_filter_split_head.vcf

cat GATK_Exon_filter_split_head.vcf |perl -lane 'if(/AF_eas=(.*)?;AF_fin.*?1000g2015aug_eas=(.*)?;CLNALLELEID/){if(($1 <= 0.001||$1 eq "\.")&&($2 <= 0.001||$2 eq "\.")){print $_}}'  >AF.vcf
# damaging1 variants : cause gene disruption (e.g., variants at splicing sites,pre-stop codon gain and stop codon loss, start codon loss, open reading frame shift) 
cat AF.vcf | perl -lane 'if(/ExonicFunc.refGene=frameshift/){print $_}if(/ExonicFunc.refGene=start/){print $_}if(/ExonicFunc.refGene=stop/){print $_}elsif(/Func.refGene=splicing/){print $_}' |awk -v OFS='\t' 'BEGIN{a="damaging1"}{print $0,".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",a}' >AF_damaging1.vcf
# damaging2 variants : damaging missense effect (predicted to have deleterious effects by at least 3 in silico programs or with CADD score ≥ 20 or DANN score ≥ 0.93)
# oNF represents the number of columns in the VCF file to be processed
# Example: if the VCF file has 1000 columns, then oNF = 1000
# oNF=1000
cat AF.vcf |perl -lane 'if(/ExonicFunc.refGene=nonsynonymous/){print $_}' |perl -lane 'if(/SIFT_pred=([^;]+);.*?Polyphen2_HDIV_pred=([^;]+);.*?Polyphen2_HVAR_pred=([^;]+);.*?MutationTaster_pred=([^;]+);.*?MutationAssessor_pred=([^;]+);.*?FATHMM_pred=([^;]+);.*?PROVEAN_pred=([^;]+);.*?M-CAP_pred=([^;]+);.*?CADD_phred=(.*)?;DANN_score=(.*)?;.*?fathmm-MKL_coding_pred=([^;]+);.*?integrated_fitCons_score=(.*)?;integrated_fitCons_rankscore/){$PP2=join("_",$2,$3);$D=join("_",$1,$4,$5,$6,$11);{print $_,"\t",$PP2,"\t",$1,"\t",$4,"\t",$5,"\t",$6,"\t",$7,"\t",$8,"\t",$9,"\t",$10,"\t",$11,"\t",$11}}'|awk -F '\t' -v OFS='\t' '{oNF=1000;if($(oNF+1)~/B_B/ || $(oNF+1)~/\._\./){print $0,$(oNF+2),"T"}else{print $0,$(oNF+2),"D"}}' |awk -F '\t' -v OFS='\t' '{oNF=1000;if($(oNF+3)~/A/ || $(oNF+3)~/D/){print $0,"D"}else{print $0,"T"}}' |awk -F '\t' -v OFS='\t' '{oNF=1000;if($(oNF+4)~/H/ || $(oNF+4)~/M/){print $0,"D",$(oNF+5),$(oNF+6),$(oNF+7)}else{print $0,"T",$(oNF+5),$(oNF+6),$(oNF+7)}}' |awk -F '\t' -v OFS='\t' '{oNF=1000;if($(oNF+8) >= 20){print $0,"D"}else{print $0,"T"}}' |awk -F '\t' -v OFS='\t' '{oNF=1000;if($(oNF+9) >= 0.93){print $0,"D"}else{print $0,"T"}}' |awk -F '\t' -v OFS='\t' '{oNF=1000;if($(oNF+11) >= 0.7){print $0,"D"}else{print $0,"T"}}' |awk -F '\t' -v OFS='\t' 'BEGIN{oNF=1000;var[1]=oNF+11;var[2]=oNF+12;var[3]=oNF+13;var[4]=oNF+14;var[5]=oNF+15;var[6]=oNF+18;var[7]=oNF+19;var[8]=oNF+20;var[9]=oNF+21;}{n=0;for(i in var){if($var[i] ~ /D/){n++}}print $0,n}' |awk -F '\t' -v OFS='\t' '{if($NF>=3){print $0}}' >AF_damaging2.vcf
cat damaging1.vcf damaging2.vcf |sort |uniq >damagingtag.vcf
